subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProvider.kt

```diff
@@ -15,199 +15,59 @@
  */
 package org.gradle.kotlin.dsl.support
 
-import org.gradle.api.artifacts.ClientModule
-import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.ModuleVersionSelector
 import org.gradle.api.artifacts.dsl.DependencyHandler
-import org.gradle.api.artifacts.dsl.RepositoryHandler
-import org.gradle.api.internal.classpath.ModuleRegistry
-
-import org.gradle.cache.CacheRepository
-import org.gradle.cache.PersistentCache
 
 import org.gradle.kotlin.dsl.embeddedKotlinVersion
 
-import java.io.File
-
-import java.net.URI
-import java.util.*
-
-
-private
-const val embeddedRepositoryCacheKeyVersion = 3
+import java.util.Properties
 
 
-class EmbeddedKotlinProvider constructor(
-    private val cacheRepository: CacheRepository,
-    private val moduleRegistry: ModuleRegistry
-) {
-
-    fun addRepositoryTo(repositories: RepositoryHandler) {
-
-        repositories.maven { repo ->
-            repo.name = "Embedded Kotlin Repository"
-            repo.url = embeddedKotlinRepositoryURI
-            repo.metadataSources { sources ->
-                sources.artifact()
-            }
-        }
-    }
+class EmbeddedKotlinProvider {
 
     fun addDependenciesTo(
         dependencies: DependencyHandler,
         configuration: String,
         vararg kotlinModules: String
     ) {
-
-        embeddedKotlinModulesFor(kotlinModules).forEach { embeddedKotlinModule ->
-            dependencies.add(configuration, clientModuleFor(dependencies, embeddedKotlinModule))
+        kotlinModules.forEach { kotlinModule ->
+            dependencies.add(configuration, kotlinModuleVersionNotationFor(kotlinModule))
         }
     }
 
     internal
-    fun pinDependenciesOn(configuration: Configuration, pinnedDependencies: Set<EmbeddedModule>) {
-        configuration.resolutionStrategy.eachDependency { details ->
-            pinnedDependencies.findWithSameGroupAndNameAs(details.requested)?.let { pinned ->
-                details.useTarget(pinned.notation)
+    fun pinEmbeddedKotlinDependenciesOn(
+        dependencies: DependencyHandler,
+        configuration: String
+    ) {
+        embeddedKotlinVersions.forEach { (module, version) ->
+            dependencies.constraints.add(configuration, module) { constraint ->
+                constraint.version {
```

Comment:
⭕️ I'd try to make this block look more like the DSL version by using `apply`:
```kotlin
    constraint.apply {
        version { it.strictly(version) }
        because("...")
    }
```

Reply:
Done